address.go

```diff
@@ -81,6 +118,37 @@ type Address interface {
 // When the address does not encode the network, such as in the case of a raw
 // public key, the address will be associated with the passed defaultNet.
 func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
+
+	// Bech32 encoded segwit addresses start with an human-readable part
+	// (hrp) followed by 1. For Bitcoin mainnet the hrp is "bc", and for
+	// testnet it is "tb". If the address string matches the hrp found in
+	// the defaultNet params, we try to decode it as a segwit address.
+	hrp := defaultNet.Bech32HRPSegwit
+	prefix := hrp + "1"
+	if strings.HasPrefix(strings.ToLower(addr), prefix) {
```

Comment:
I added the utility method in https://github.com/btcsuite/btcd/pull/968, and updated the code here to use it.